### DISCUSSION TITLE
Make it easier to discover categories and keywords

### DISCUSSION THREAD

**Discussion by e00E** [rust-lang/crates.io#4319] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
I am looking for crates related to websockets so I enter websocket into the search. [This](https://crates.io/search?q=websocket) does not show me some of the most popular crates like tungstenite and socketto because the default search is by relevance and those crates do not contain this in their name (presumably this is the cause).
Only when I click on one of the crates that show up in the search I discover the websocket keyword and category. Using either I immediately find the most popular crates.
Note that I could have also changed the first search to go by most popular instead of relevance for a similar result.

**Describe the solution you'd like**
When searching for a word that has been used as a keyword or is a category the page should prominently display this at the top. Something like "Would you like to search in the keyword/category instead?".
This makes it a lot easier to discover keywords and categories as otherwise you will not see them until you click on a crate that uses them.

**Describe alternatives you've considered**
An alternative for users is to use the https://crates.io/keywords and https://crates.io/categories pages that are linked at the frontpage. However these links are easy to overlook and once you did find them you still cannot search for keywords and categories.

**Comment by Yummy-Yums** [comment]
+1 on this , you can use  [scaladex](https://index.scala-lang.org/) as inspiration
